File: 204countPrimes/countPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n < 2) return 0;
        else
        {
            std::vector<int> result(n, 1);
            result.front() = result.at(1) = 0;
            for (std::size_t i{2}; i != static_cast<int>(std::sqrt(n)) + 1; ++i)
                for (std::size_t j{std::pow(i, 2)}; j < n; j += i) result.at(j) = 0;
            return std::accumulate(std::cbegin(result), std::cend(result), 0);
        }
    }
};

File: 204countPrimes/countPrimes.java
class Solution {
    public int countPrimes(int n) {
        if (n < 2) return 0;
        else
        {
            final List<Integer> result = Collections.nCopies(n, 1).stream().collect(Collectors.toList());
            result.set(0, 0);
            result.set(1, 0);
            IntStream.range(2, (int)Math.sqrt(n) + 1).forEach(i -> {
                for (int j = (int)Math.pow(i, 2); j < n; j += i) result.set(j, 0);});
            return result.stream().mapToInt(Integer::intValue).sum();
        }
    }
}
